src/views/CreateBox/CreateBox.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from "react";
+import { useMutation, useQuery } from "@apollo/react-hooks";
+import { useForm } from "react-hook-form";
+import { useLocation, Link } from "react-router-dom";
+import AuthContext from "../../AuthContext";
+import { NewBoxType, LocationState, AuthObjectType } from "../../utils/Types";
+import { USER, CREATE_BOX } from "../../utils/queries";
+import { locationOptions } from "../../utils/locationOptions";
+import { emptyBox } from "../../utils/emptyBox";
+
+export default function CreateBox() {
+  const authObject: AuthObjectType = React.useContext(AuthContext);
+  const { email } = authObject.idTokenPayload;
+
+  // NOTE: getting the user will likely eventually have to be done in a more global-place,
+  // maybe App and make a context for it? If that happens, consider using useLazyQuery for it
+  // https://www.apollographql.com/docs/react/data/queries/#executing-queries-manually
+  const { loading: queryLoading, error: queryError, data: queryData } = useQuery(USER, {
+    variables: { email },
+  });
+
+  const [createBoxMutation, { loading: mutationLoading, error: mutationError }] = useMutation(
+    CREATE_BOX,
+  );
+
+  const location: LocationState = useLocation();
+  const qrUrl: string = location?.state?.qr;
+  // FOR TESTING
```

Comment:
Here I have included a valid QR barcode so you don't have to generate one. If you're testing this out locally, you can comment out the top qrBarcode and uncomment the bottom one, and then it won't complain about whatever random qr code not being in the database.